airflow/api_connexion/security.py

```diff
@@ -37,3 +37,32 @@ def decorated(*args, **kwargs):
         return function(*args, **kwargs)
 
     return cast(T, decorated)
+
+
+def requires_access(permissions: Sequence[Tuple[str, str]]) -> Callable[[T], T]:
```

Comment:
should we combine @security.requires_access and @security.requires_authentication into one decorator instead? will there be a case where one would only use require_access without requires_authentication? this doesn't make sense semantically right?

Reply:
@houqp we do that, but there are times when authentication is needed, but we don't tie it to a specific permission. Such as the Health and Configs endpoints. We could add a semi-arbitrary permission for those cases (`('Health', 'can_read')`), or similar with reading configs. Or we could allow `requires_access()` to work without any permissions, in which case it only requires authentication.

What do you think? @mik-laj curious about your thoughts as well.

Reply:
For endpoints that need to be guarded by authentication but not authorization, one solution we can adopt is to pass in an empty permissions list, or `None`, then in the decorator skip the role check entirely if that's the case.

Reply:
yeah that isnt particularly difficult, but I think it complicates the interface somewhat. I think I'll just create resources that aren't based on a specific model. So 'Health' or 'Configuration', just to keep the format consistent

Reply:
Config endpoint should have authentication and authorization.
Health and version endpoint don't need any protection. There are no sensitive data there and for operational reasons it is better not to add them. This will greatly facilitate monitoring, autodiscovery and API client development.

Reply:
Thanks. Good to know. I've removed the Health and Version authorization.